mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -524,6 +525,65 @@ class BooleanParam(parent: String, name: String, doc: String) // No need for isV
   }
 }
 
+/**
+ * :: DeveloperApi ::
+ * Specialized version of `Param[String]` for Java.
+ */
+@DeveloperApi
```

Comment:
I am leaning towards just not doing this. I think it adds complexity for the same functionality.

Reply:
this `normalize` is only added to `StringParam` extending `Param[String]`, so it will affect other type of params. 
OK, I also feel it somewhat complex.